src/cpp/rtps/xmlparser/XMLParser.cpp

```diff
@@ -1771,29 +1771,54 @@ XMLP_ret XMLParser::fillDataNode(
     addAllAttributes(p_profile, participant_node);
 
     uint8_t ident = 1;
-    tinyxml2::XMLElement* p_element = p_profile->FirstChildElement(DOMAIN_ID);
-    if (nullptr != p_element)
+    tinyxml2::XMLElement* p_element;
+    tinyxml2::XMLElement* p_aux0 = nullptr;
+    const char* name = nullptr;
+    std::unordered_map<std::string, bool> tags_present;
+
+    /*
+     * The only allowed elements are <domainId> and <rtps>
+     *   - The min occurrences of <domainId> are 0, and its max is 1; look for it.
+     *   - The min occurrences of <rtps> are 0, and its max is 1; look for it.
+     */
+    for (p_element = p_profile->FirstChildElement(); p_element != nullptr; p_element = p_element->NextSiblingElement())
     {
-        // domainId - uint32Type
-        if (XMLP_ret::XML_OK != getXMLUint(p_element, &participant_node.get()->domainId, ident))
+        name = p_element->Name();
+        if (tags_present[name])
```

Comment:
`operator[]` inserts the element if not exists. I suggests:

```suggestion
        if (tags_present.contains(name))
```

Reply:
Good point! I've changed it to a set